### ISSUE TITLE
Make the warp core require lots of power.

### ISSUE THREAD

**Issue by Gliese-832-c** [zmaster587/AdvancedRocketry#1738]
The warp core should require a lot of power. In Star Trek dilithium is actually only used to stabilize it, since antimatter doesn't explode on contact with it (don't ask me how that works, I'm not Gene Rodenberry). The actual power comes from a matter-antimatter reactor, and I feel like the bottom gold block in the warp core multiblock could be replced with a power input plug or something. Possibly some kind of advanced power input plug. (Also, the machine blocks should be replaced with advanced machine blocks imo). It should require let's say 100000 RF/t when in flight (this number should be configurable though), and if the power runs out while in flight, it will explode (the explosion should be very big). To avoid having to add things like being stranded in interstellar space, let's say that the ship will still arrive at the destination, and the core will explode there.

TL;DR: The warp core should use looots of power.

**Comment by Gliese-832-c** [comment]
Though if @zmaster587 want's to take the time and wants the cores to explode instantly, then the spaceship could become stranded in interstellar space, and some kind of damaged space ship icon would pop up in the warp controller/planet selector GUI. If you have another ship/are another player, you could go there (you would still have to fly there with a rocket, but the skybox will be replaced with the same broken ship icon). You can then help the player if they are still alive, or loot the ship if no one's there (afaik you respawn on earth upon death).

I don't know how hard it would be to implement dynamic warp controller/planet selector map changing, so this is not part of my main suggestion.

**Comment by Silfryi** [comment]
There's reason not to do this, because keeping compatibility with Vanilla and AdvR only means it could only consume meager amounts of power, like the machines. This would work better for other drives, when they're implemented.

**Comment by Gliese-832-c** [comment]
Then the number could be made low, but configurable with the possiblity to be very high (the maximum could be MAX_INT), because in my modpack there are plenty of ways to get a lot of power, fission, fusion, resonating reactor, spamming a lot of highest tier NuclearCraft solar panels, beaming the power using quantum entangloporters from Mekanism, I'm pretty sure you could come up with a way to power the warp core. Also vanilla and AR is impossible anyways, because according to the actual item name the energy pipes (and the fluid and data pipes too) deprecated, so it's impossible to play with only vanilla and AR. Plus, an extreme vast majority likely plays the mod with other mods, because only playing with AR is kind of difficult, I mean, the only power gen options are coal generators which produce 40 RF/t, and solar panels which produce 1 RF/t, with no way to transfer that power.

**Comment by C1ffisme** [comment]
I actually think it does make sense for the reactor to require a little bit of external power, possibly configurable by a config. Sort of like a gasoline engine: you need a little bit of electricity to spark the spark-plugs even if the engine runs on gasoline.

That sensical argument aside, I don't really think it's worthwhile gameplay-wise to add an energy requirement in addition to a fuel requirement.

I think a more worthwhile cause, specifically for modpack creators, would be to:

- Make the Warp Core accept any of the items in a list in the config and
- Allow the Warp Core to be disabled in favor of other new engines, such as...
- The Hyperdrive, which runs on a configurable amount of RF and can be disabled in the config or
- The Exotic Matter drive, which runs on any of the fluids in a list in the config and can be disabled in the config. (The default fluid would be Hydrogen for simplicity.)

(Possibly in addition to this, other drives that run on both RF and fuel, RF and fluid, fluid and fuel could be constructed.)

Lastly, I also don't approve of anything making a "big explosion" on a space station. This is just asking for something to go wrong. (Player gets stranded in space station dimension, stranded in orbit around a planet without materials needed to rebuild warp core, etc.)

**Comment by MuteTiefling** [comment]
I'd agree that it makes sense for it to use power. However, I'm not sure that's best left to the mod itself. 

Personally, in the pack I'm working on I've simply opted to make them power hungry to craft. On the order of several hundred million RF. Of course, this only works because my pack also has that level of power gen available.

**Comment by fimzks** [comment]
i don't think a big explotion or even getting stranded, at least for the first releases of this feature(that I'd love to see) i think the warp core should have a "charging bar" for an internal "capacitor" so when its fully loaded you can jump to another location, that way you don't need to work in all the stranded feature at least for some time, also you could make the capacitor a slot so you can insert better capacitors to travel larger distances. Then if the stranded mechanic gets implemented you can make the core with the capacitor slot a higher tear of core so you dont get stranded
